examples/gitserver/README.md

```diff
@@ -1,16 +1,207 @@
+
+Quick Start
+-----------
+
+Prerequisites:
+
+* You have an OpenShift v3 server running
+* You are logged in and have access to a project
+* You have the `gitserver.yaml` from this directory
+* You can create externally accessible routes on your server
+
+### Deploy the Git Server
+
+1. Create the Git Server
+
+    ```sh
+    $ oc create -f gitserver.yaml
+    ```
+
+2. Grant `edit` access to the `git` service account
+
+    ```sh
+    $ oc policy add-role-to-user edit -z git
+    ```
+
+
+### Push code to the Git Server
+
+Code repositories can be initialized on the Git Server by either doing a 'git push' or
+a 'git clone' of the repository.
+
+
+1. Find your git server's URL
+
+   If you have a working router, determine the host name of your git server by getting its route:
+
+   ```sh
+   $ oc get route git
+   ```
+
+   In this case, the URL of your git server will be the host name used by the route; something like: 
+   
+   ```
+   http://git-myproject.infra.openshift.com
+   ```
+  
+   Alternatively, if your router is not functional, you can port-forward the git-server pod to your local machine.
+   In a separate shell window, execute the following: (You must leave the port-forward command running
+   for the port to be forwarded to your machine)
+
+   ```
+   $ oc get pods | grep git           ## get the git server pod
+   $ oc port-forward -p PODNAME 8080  ## start port-forward where PODNAME is the git server pod
+   ```
+
+   In this case, the URL of your git server will be your local host:
+
+   ```
+   http://localhost:8080
+   ```
+
+
+2. Setup your credentials
+
+   By default the Git Server will allow users or service accounts that can create pods in
+   the project namespace to create and push code to the Git Server. The easiest way to 
+   provide credentials to the Git Server is by using a custom credential helper that will 
+   send your OpenShift token by default to the server.
+   ```sh
+   $ git config --global credential.http://gitserver-myproject.infra.openshift.com.helper \
+         '!f() { echo "username=$(oc whoami)"; echo "password=$(oc whoami -t)"; }; f'
+   ```
+
+   **NOTE:** the config key is `credential.[git server URL].helper`
+
+3. Push a repository to your git server
+
+   In an existing repository, add a remote that points to the git server and push to it
+   
+   ```
+   # clone a public repository
+   $ git clone https://github.com/openshift/ruby-hello-world.git
+
+   # add a remote for your git server
+   $ cd ruby-hello-world
+   $ git remote add openshift http://git-myproject.infra.openshift.com/ruby-hello-world.git  
+
+   # push the code to the git server
+   $ git push openshift master
+   ```
+
+   **NOTE:** the ruby-hello-world.git repository does not exist before running these commands. 
+   By pushing to it, you are creating it in the Git Server.
+
+   On push the git server will invoke new-app on the code and create artifacts for it in 
+   OpenShift.
```

Comment:
so if i don't want new-app invocation because i've already setup my buildconfig/deploymentconfig, how do i get my repo created w/o this happening?